src/tensor/aggregate.nim

```diff
@@ -18,19 +18,25 @@ import  ./backend/memory_optimization_hints,
         ./higher_order_foldreduce,
         ./math_functions,
         ./accessors,
+        ./private/p_empty_tensors,
         math
 import complex except Complex64, Complex32
 
 # ### Standard aggregate functions
 # TODO consider using stats from Nim standard lib: https://nim-lang.org/docs/stats.html#standardDeviation,RunningStat
 
+# Note: for aggregate that returns scalar, if the tensor is empty,
+#       Numpy seems to return the neutral element, do we want that?
+
```

Comment:
From a purely mathematical point of view it does make sense. I see your point thought that it can introduce subtle bugs for cases where an empty tensor is actually the result of something that went wrong. By returning the identity that information is lost. 

But since the only real alternative I see is to raise an exception, I'm not sure. I'll think about it some more.